### PR TITLE
Add CacheAwareInterface, CacheAwareTrait and NullCache.

### PR DESCRIPTION
**Pull request by drupol**
Hi,

This PR is related to this [mailing-list post](https://groups.google.com/forum/#!topic/php-fig/hZmfnwKqf1I) and originates from [Psr/simple-cache #10](https://github.com/php-fig/simple-cache/pull/10) and [Psr/simple-cache #11](https://github.com/php-fig/simple-cache/pull/11).

Should we create a new repo for this or include it in this repo, the question is open and to be discussed.

### CODE REVIEW

**Comment by harikt** [comment]
All looks good. Only thing I specially notice is inconsistency in the coding standard.

**Comment by drupol** [comment]
I'll take care of that today.

**Comment by drupol** [comment]
Ready for review now.

File: src/NullCache.php [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    {
+        return $default;
+    }
+
+    /**
+     * {inheritdoc}
+     */
+    public function getMultiple($keys, $default = null)
+    {
+        return $default;
```
Comment (nicolas-grekas) [inline src/NullCache.php]:
```
        foreach ($keys as $key) {
            yield $key => $default;
        }
```

File: src/NullCache.php [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    {
+        return true;
+    }
+
+    /**
+     * {inheritdoc}
+     */
+    public function setMultiple($values, $ttl = null)
+    {
+        return true;
```
Comment (nicolas-grekas) [inline src/NullCache.php]: return false

File: src/NullCache.php [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    {
+        return $default;
+    }
+
+    /**
+     * {inheritdoc}
+     */
+    public function set($key, $value, $ttl = null)
+    {
+        return true;
```
Comment (nicolas-grekas) [inline src/NullCache.php]: return false

File: src/NullCache.php (hunk 1 above)
Comment (nicolas-grekas) [reply src/NullCache.php]: or a PHP5.4 compatible version of it as 5.4 is advertised as supported...

Comment (drupol) [reply src/NullCache.php]: I've replaced it with something very simple:
```php
        $return = [];

        foreach ($keys as $key) {
            $return[$key] = $default;
        }

        return $return;
```

Is it ok ?

Comment (nicolas-grekas) [reply src/NullCache.php]: yes

**Comment by danopz** [comment]
Should be https://github.com/php-fig/simplecache-util

**Comment by drupol** [comment]
I've opened a discussion on the mailing list about it: https://groups.google.com/forum/#!topic/php-fig/hZmfnwKqf1I

Has this already been decided then ?

**Comment by drupol** [comment]
I've added a PR against the new repo: https://github.com/php-fig/simplecache-util/pull/1

**Pull request closed without merging**